### PR TITLE
Backport Getting started with Kubernetes

### PR DESCRIPTION
**Pull request by benarent**
Backports Kubernetes Getting started changes.  https://github.com/gravitational/teleport/pull/5981/

### CODE REVIEW

File: docs/pages/kubernetes-access/getting-started.mdx [hunk 1]
```diff
@@ -151,7 +151,7 @@ Teleport will respond back with a discovery info:
 ```bash
 $ curl https://tele.example.com/webapi/ping
 
-{"server_version":"5.0.0-dev","min_client_version":"3.0.0"}
+{"auth":{"type":"local","second_factor":"otp"},"proxy":{"kube":{"enabled":true,"listen_addr":"0.0.0.0:3026"},"ssh":{"listen_addr":"[::]:3023","tunnel_listen_addr":"0.0.0.0:3024","public_addr":"tele.example.com:443"},"db":{}},"server_version":"6.0.1","min_client_version":"3.0.0"}
```
Comment (klizhentas) [inline docs/pages/kubernetes-access/getting-started.mdx:154]: I removed those on purpose to focus attention on the key details

File: docs/pages/kubernetes-access/getting-started.mdx [hunk 2]
```diff
@@ -178,13 +178,13 @@ Create the role and add a user:
 
 ```bash
 # To create a local user, we are going to run Teleport's admin tool tctl from the pod.
-$ POD=$(kubectl get po -l app=teleport-cluster -o jsonpath='{.items[0].metadata.name}')
```
Comment (klizhentas) [inline docs/pages/kubernetes-access/getting-started.mdx:181]: this is not a typo, this should work

Comment (benarent) [reply docs/pages/kubernetes-access/getting-started.mdx:181]: TIL Kubernetes supports `po` as an [abbreviation](https://kubernetes.io/docs/reference/kubectl/overview/#syntax) of `pod` I'll double check that is wasn't an issue with something else

File: docs/pages/kubernetes-access/getting-started.mdx (hunk 1 above)
Comment (benarent) [reply docs/pages/kubernetes-access/getting-started.mdx]: Updated to the shorter version but updated the `server_version` to 6.0.0

File: docs/pages/kubernetes-access/getting-started.mdx (hunk 2 above)
Comment (benarent) [reply docs/pages/kubernetes-access/getting-started.mdx:181]: I've reverted this change.

**Comment by benarent** [comment]
@klizhentas I've updated this guide to use the new preset roles. PTAL and let me know what you think.

File: docs/pages/kubernetes-access/getting-started.mdx [hunk 3]
```diff
@@ -151,42 +151,24 @@ Teleport will respond back with a discovery info:
...
 
-# Create a role
-$ kubectl exec -i ${POD?} tctl -- create -f < member.yaml
-
 # Generate an invite link for the user.
-$ kubectl exec -ti ${POD?} tctl -- users add alice --roles=member
+$ kubectl exec -ti ${POD?} -- tctl users add teleport-admin --roles=editor,access --kubernetes_groups=system:masters --logins=k8s-keep-this-login
 
-User "alice" has been created but requires a password. Share this URL with the user to
+User "teleport-admin" has been created but requires a password. Share this URL with the user to
```
Comment (benarent) [inline docs/pages/kubernetes-access/getting-started.mdx:171]: I updated this so it's the same user and flow as our getting started guide.

File: docs/pages/kubernetes-access/getting-started.mdx [hunk 4]
```diff
@@ -151,42 +151,24 @@ Teleport will respond back with a discovery info:
...
 ```bash
 # To create a local user, we are going to run Teleport's admin tool tctl from the pod.
 $ POD=$(kubectl get po -l app=teleport-cluster -o jsonpath='{.items[0].metadata.name}')
 
-# Create a role
-$ kubectl exec -i ${POD?} tctl -- create -f < member.yaml
-
 # Generate an invite link for the user.
-$ kubectl exec -ti ${POD?} tctl -- users add alice --roles=member
+$ kubectl exec -ti ${POD?} -- tctl users add teleport-admin --roles=editor,access --kubernetes_groups=system:masters --logins=k8s-keep-this-login
```
Comment (benarent) [inline docs/pages/kubernetes-access/getting-started.mdx:169]: I've added `--kubernetes_groups=system:masters --logins=k8s-keep-this-login` since the new default roles don't have anything set. If there is role with less privilege you think I should use please let me know.

**Comment by benarent** [comment]
( Once merged I'll create a new PR to add these changes back into master )

File: docs/pages/kubernetes-access/getting-started.mdx [hunk 5]
```diff
@@ -151,42 +151,24 @@ Teleport will respond back with a discovery info:
...
-metadata:
-  name: member
-spec:
-  allow:
-    # This field is used for SSH logins. You have to keep 'logins' as a non-empty random value
-    # for Kubernetes to work until we fix it.
-    logins: ['keep-this-value-here']
-    kubernetes_groups: ["system:masters"]
-```
+Let's create a local admin `teleport-admin` who has access to Kubernetes group `system:masters`.
```
Comment (klizhentas) [inline docs/pages/kubernetes-access/getting-started.mdx:160]: Why did you remove this example?

File: docs/pages/kubernetes-access/getting-started.mdx (hunk 4 above)
Comment (klizhentas) [reply docs/pages/kubernetes-access/getting-started.mdx:169]: Why that change? The role I had above was a bit simpler I think

File: docs/pages/kubernetes-access/getting-started.mdx (hunk 5 above)
Comment (benarent) [reply docs/pages/kubernetes-access/getting-started.mdx:160]: I removed it to use the preset role instead `kubectl exec -ti ${POD?} -- tctl users add teleport-admin --roles=editor,access --kubernetes_groups=system:masters --logins=k8s-keep-this-login` and to require one less YAML file for setup.

File: docs/pages/kubernetes-access/getting-started.mdx (hunk 4 above)
Comment (benarent) [reply docs/pages/kubernetes-access/getting-started.mdx]: I've changed it back to how it was before.

File: docs/pages/kubernetes-access/getting-started.mdx (hunk 5 above)
Comment (benarent) [reply docs/pages/kubernetes-access/getting-started.mdx]: ( reverted this in the latest commit )

File: docs/pages/kubernetes-access/getting-started.mdx [hunk 6]
```diff
@@ -151,13 +151,12 @@ Teleport will respond back with a discovery info:
...
 
-{"server_version":"5.0.0-dev","min_client_version":"3.0.0"}
+{"server_version":"6.0.0","min_client_version":"3.0.0"}
 ```
 
 ## Step 2/3 Create a local admin
 
-Local users are a reliable fallback for cases when the SSO provider is down.
-Let's create a local admin `alice` who has access to Kubernetes group `system:masters`.
+Local users are a reliable fallback for cases when the SSO provider is down. Let's create a local admin `alic`e` who has access to Kubernetes group `system:masters`.
```
Comment (klizhentas) [inline docs/pages/kubernetes-access/getting-started.mdx:159]: Extra tick

**Comment by benarent** [comment]
@klizhentas I've completed a dry run and head a few minor edits at the end.  I had to set context again, after logging into Teleport, as without this I was running into this error.

```
➜  ~ POD=$(kubectl get po -l app=teleport-cluster -o jsonpath='{.items[0].metadata.name}')
error: error executing jsonpath "{.items[0].metadata.name}": Error executing template: array index out of bounds: index 0, length 0. Printing more information for debugging the template:
	template was:
		{.items[0].metadata.name}
	object given to jsonpath engine was:
		map[string]interface {}{"apiVersion":"v1", "items":[]interface {}{}, "kind":"List", "metadata":map[string]interface {}{"resourceVersion":"", "selfLink":""}}
```

**Review by klizhentas — APPROVED** [review]
